main.rb

```diff
@@ -0,0 +1,75 @@
+require_relative './app'
+
+def create_a_person
+  puts 'Do you want to create (1) a Student or (2) a Teacher?'
+  selected = gets.chomp.to_i
+  print 'Age: '
+  age = gets.chomp
+  print 'Name: '
+  name = gets.chomp
+  case selected
+  when 1
+    print 'Has Parent Permission (Y/N): '
+    parent_permission = gets.chomp
+    create_student(age, name, parent_permission, 'nil')
+  when 2
+    print 'Specialization: '
+    specialization = gets.chomp
+    create_teacher(age, name, specialization)
+  end
+  main
+end
+
+def create_a_book
+  print 'Title: '
+  title = gets.chomp
+  print 'Author: '
+  author = gets.chomp
+  create_book(title, author)
+  main
+end
+
+def create_a_rental
+  puts "Select the person from the following list by number (not id)\n\n"
+  @person_list.each_with_index do |person, index|
+    if person.is_a?(Teacher)
+      puts "[Teacher] Name: #{person.name}, ID: #{person.id} Age: #{person.age}"
+    else
+      puts "#{index}. [Student] Name: #{person.name}, ID: #{person.id} Age: #{person.age}"
+    end
+  end
+  selected_person = gets.chomp.to_i
+  puts "Select the book from the following list\n\n"
+  @books_list.each_with_index do |book, index|
+    puts "#{index}. [Book] Name: #{book.title} Author: #{book.author}"
+  end
+  selected_book = gets.chomp.to_i
+  puts 'Date'
+  date = gets.chomp
+  create_rental(date, @person_list[selected_person], @books_list[selected_book])
+end
+
+def list_all_rentals
+  print 'ID of person: '
+  rental_id = gets.chomp.to_i
+  list_rentals(rental_id)
+end
+
+puts 'Welcome to the School Library App'
+
+def main
+  puts "Please choose an option by entering an option\n"
+  puts '1. List all Books'
+  puts '2. List all People'
+  puts '3. Create a Person'
+  puts '4. Create a Book'
+  puts '5. Create a Rental'
+  puts '6. List all rentals for a given person id.'
+  puts '7. Exit'
+  option = gets.chomp.to_i
+
+  choose_option(option)
```

Comment:
- [ ] The first error I got while testing option 4 - **Create book** - is seen below 👇🏾 

   ![image](https://github.com/Mutai-Gilbert/School-Library/assets/84156714/75e50aec-b9d1-4e9d-80f2-156073a85ad4)

   This simply means the method `choose_option` is undefined in the `main.rb` file. I notice you have it in the `app.rb` file, but in your `main.rb` file, you do not have direct access to it, except if you instantiate the Library class or use `self.choose_option`. Please see this repo for a clue to how to set things up in the main folder. You may not follow all but take a cue to what is expected. Currently, you have a lot of errors popping out.

- [ ] Then a second error after kept checking, 👇🏾 
Also undefined method. 

  ![image](https://github.com/Mutai-Gilbert/School-Library/assets/84156714/5ca692ae-35db-4b1a-992f-7b2f3862e23a)



Kindly fix all this and ensure your Library app works as expected.